src/is.js

```diff
@@ -125,6 +125,33 @@ export function isWindowObject(value) {
     return value != null && typeof value === 'object' && 'setInterval' in value;
 }
 
+/**
+ * Returns whether all values in the 
+ * array/object are equal to a given 
+ * value
+ * @param {*} value the array or object to check
+ * @param {*} other the value or function to match
+ */
+export function matchesAll(value, other) {
```

Comment:
Please rename `other` to a more significant identifier. 
I suggest `matcher`. Feel free to suggest something better.